src/modi/mod.rs

```diff
@@ -335,3 +310,40 @@ impl<'a> FallibleIterator for FileIterator<'a> {
         }
     }
 }
+
+/// Named reference to a [`Module`].
+///
+/// The name stored in the [`StringTable`] corresponds to the name of the module as returned by
+/// [`Module::module_name`].
+///
+/// [`Module`]: struct.Module.html
+/// [`Module::module_name`]: struct.Module.html#method.module_name
+/// [`StringTable`]: struct.StringTable.html
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct ModuleRef(pub StringRef);
```

Comment:
This is a bit ugly, because it currently requires the user to do `module_ref.0.to_string_lossy(&strtab)`. 

Potentially, we could provide a stateful lookup table based on the DBI module stream that allows something like `fn get(&mut self, module_ref: ModuleRef, strtab: &StringTable<'_>) -> Result<Option<_>>`. At the moment it's unclear to me, whether one might want the index of the module, the parsed header, or something completely different, though.